stream_consumer.go

```diff
@@ -284,52 +284,56 @@ func (se *streamEndpoint) reconnectWhileNotStopped(c *consumer, streamName strin
 		//without this hack we do not know if the stream is really connected
 		mds, err := st.Header()
 		if err == nil && mds != nil {
-
-			if config.OnConnected != nil {
-				config.OnConnected(streamName)
+			var cs connectionStatus
+			if mds.Get("expectHello") != nil && len(mds.Get("expectHello")) > 0 {
```

Comment:
why use the the header for this?

Reply:
because we might be connected to a server using the old version, that will not send a hello message, in this case we should not discard the first message.

Reply:
once the migration is done we can remove the header logic